packages/frontend/src/redux/slices/ledger/index.js

```diff
@@ -1,14 +1,192 @@
+import { createAsyncThunk, createSlice, current } from "@reduxjs/toolkit";
+import set from 'lodash.set';
+import unset from 'lodash.unset';
 import { createSelector } from "reselect";
 
+import { HIDE_SIGN_IN_WITH_LEDGER_ENTER_ACCOUNT_ID_MODAL } from "../../../config";
+import { showAlertToolkit } from "../../../utils/alerts";
+import { setLedgerHdPath } from "../../../utils/localStorage";
+import { wallet } from "../../../utils/wallet";
+import refreshAccountOwner from "../../sharedThunks/refreshAccountOwner";
+
 const SLICE_NAME = 'ledger';
 
+export const LEDGER_MODAL_STATUS = {
+    CONFIRM_PUBLIC_KEY: 'confirm-public-key',
+    CONFIRM_ACCOUNTS: 'confirm-accounts',
+    ENTER_ACCOUNTID: 'enter-accountId'
+};
+
+const initialState = {
+    modal: {}
+};
+
+const getLedgerAccountIds = createAsyncThunk(
+    `${SLICE_NAME}/getLedgerAccountIds`,
+    async ({ path }) => await wallet.getLedgerAccountIds({ path }),
+    showAlertToolkit({ onlyError: true })
```

Comment:
Just trying to wrap my head around the new need for `showAlertToolkit`... Is this the first time we're decoupling an async thunk from the `readyStatePromise` middleware?

Reply:
Yes, I mean, these are the first async actions that are using `globalAlert` that are refactored. I started refactoring from the simplest reducers, so there ware no case like that before.